src/Helpers/IPFS.ts

```diff
@@ -7,25 +7,35 @@ import { minutesToMiliseconds } from './Time'
 
 const getHash = prop('Hash')
 
+interface FetchOptions {
+  readonly timeout?: number
+}
+
 export interface IPFSConfiguration {
   readonly url?: string
 }
 
-interface FetchOptions {
-  readonly timeout?: number
+type addText = (config?: FetchOptions) => (text: string) => Promise<string>
+type cat = (options?: FetchOptions) => (hash: string) => Promise<string>
+
+export interface IPFS {
+  addText: addText,
+  cat: cat
 }
 
-export const IPFS = ({
+type createIPFS = (config?: IPFSConfiguration) => IPFS
+
+export const createIPFS: createIPFS = ({
```

Comment:
It seems I could have left this named `IPFS`, allowing IPFS to represent both the returned object and the factory.

```suggestion
export const IPFS: createIPFS = ({
```

TypeScript seemed to have no issues with that and it worked fine, but I couldn't find any TS documentation on it so I was unsure if it was just a bug.

Reply:
> It seems I could have left the factory named IPFS, allowing IPFS to represent both the returned object and the factory. 

Yep, that's a common pattern in TS. TS things and JS things can share the same name since TS things vanish after the build step, roughly speaking. A const and an interface having the same name isn't an issue.

I liked the idea of naming the factories after their returned object in PascalCase.

Maybe we can avoid having a function `IPFS` of type `createIPFS` too, by skipping the separate type declaration:

```ts
export const IPFS = ({
  url = 'http://localhost:5001',
}: IPFSConfiguration = {}): IPFS => {
```

Sometimes separating the type declaration from the implementation doesn't add that much value IMO.

I'm fine either way though.